content/actions/reference/context-and-expression-syntax-for-github-actions.md

```diff
@@ -109,13 +109,22 @@ The `env` context contains environment variables that have been set in a workflo
 
 The `env` context syntax allows you to use the value of an environment variable in your workflow file. If you want to use the value of an environment variable inside a runner, use the runner operating system's normal method for reading environment variables.
 
-You can only use the `env` context in the value of the `with` and `name` keys, or in a step's `if` conditional. For more information on the step syntax, see "[Workflow syntax for {% data variables.product.prodname_actions %}](/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idsteps)."
+You can only use the `env` context in the value of certain keys in a step:
+
+- `if`
+- `name`
+- `run`
+- `with` (including `with.args` and `with.entrypoint`)
+- `env`
+- `continue-on-error` (required conversion from string to boolean)
+- `timeout-minutes` (requires conversion from string to number)
+
+ For more information on the step syntax, see "[Workflow syntax for {% data variables.product.prodname_actions %}](/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idsteps)."
```

Comment:
Thanks for testing this! I really respect the work you put into validating this. ✨ 

If new keys are added in the future, it seems that the default behavior is to allow it to use the `env` context. To make it easier to keep the docs up-to-date, I recommend documenting the keys that won't accept the `env` context instead of the keys that do accept it. To do that, you could add this to line 110, after "The `env` context syntax allows you to use the value of an environment variable in your workflow file.":

> You can use the `env` context in the value of any key of a step except for the `id`, `uses`, or `env` keys. For more information on the step syntax, see "[Workflow syntax for {% data variables.product.prodname_actions %}](/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idsteps)."

And then can remove lines 112-122.

What do you think?

Reply:
Yeah, it's probably better this way around from a maintenance perspective and to keep it succinct. Adjusted the PR as suggested, with some slight modifications.

This is my test workflow BTW:

```yaml
env:
  JOB_NAME: Name defined at job level
  JOB_OS: windows-latest
  JOB_ID: foo
  JOB_ACTION: actions/github-script@v3
  JOB_TIMEOUT: 1
  JOB_BOOL: true
  JOB_WITH: |
    { "script": "console.log('with block supplied as JSON string')" }
  JOB_STEP: |
    {"name": "env as step", "run": "echo Entire step defined as string"}
  JOB_ENTRY: /custom.sh
  JOB_ARGS: FirstArgument SecondArgument?

#name: ${{ env.JOB_NAME }} # error
name: Env context

on:
  push:
    branches:
      - main

jobs:
  job-level:
    #name: ${{ env.JOB_NAME }} # error
    name: Job 1
    #runs-on: ${{ env.JOB_NAME }} # error
    runs-on: ubuntu-latest
    steps:
      - name: env context in run
        run: echo ${{ env.JOB_NAME }} # works
      - name: env var in action
        uses: actions/github-script@v3
        with:
          script: |
            console.log('process.env.JOB_OS: ' + process.env.JOB_OS)
          # works, set via job level env
      - name: env context in action
        uses: actions/github-script@v3
        env:
          my_os: ${{ env.JOB_OS }} # works
        with:
          script: |
            console.log('process.env.my_os: ' + process.env.my_os)
      - name: env in id (set output)
        #id: ${{ env.JOB_ID }} # error
        id: foo
        run: |
          echo "::set-output name=result::bar"
      - name: env in id (read output)
        run: |
          echo "result = ${{ steps[env.JOB_ID].outputs.result }}"
      - name: env in uses
        #uses: ${{ env.JOB_ACTION }} # error
        uses: actions/github-script@v3
        with:
          script: console.log("${{ env.JOB_ACTION }}") # works
      - name: env in with
        uses: actions/github-script@v3
        with: ${{ fromJSON(env.JOB_WITH) }} # works
      #- ${{ fromJSON(env.JOB_STEP) }} # error
      - name: env in with.args
        uses: actions/hello-world-docker-action@v1
        with:
          entrypoint: /bin/echo
          args: ${{ env.JOB_ARGS }} # works
      - name: env in with.entrypoint
        uses: actions/hello-world-docker-action@v1
        continue-on-error: true
        with:
          who-to-greet: 'Mona the Octocat'
          entrypoint: ${{ env.JOB_ENTRY }} # works
      - name: env in timeout
        continue-on-error: true
        timeout-minutes: ${{ fromJSON(env.JOB_TIMEOUT) }} # works
        run: |
          echo sleeping for 2 minutes
          sleep 120
          echo done
      - name: env in continue
        continue-on-error: ${{ !!env.JOB_BOOL }} # works
        run: |
          exit 1
```